src/simplify_algebra.cpp

```diff
@@ -1367,6 +1369,9 @@ struct find_split_reshape
         auto slc_lens     = slc->get_shape().lens();
         auto slc_dim_size = std::accumulate(
             slc_lens.begin() + axis, slc_lens.end(), 1, std::multiplies<std::size_t>());
+        auto input_lens   = input->get_shape().lens();
+        auto input_size   = input->get_shape().elements();
+        auto slc_axis_len = input_lens[axis];
```

Comment:
I think matcher should also check if there is only axis that is being sliced. If multiple axes are sliced then it should return (perhaps we can do better).

Reply:
added